src/consumer.rs

```diff
@@ -576,6 +583,13 @@ pub async fn from_substrait_rex(
             Some(LiteralType::Fp64(f)) => {
                 Ok(Arc::new(Expr::Literal(ScalarValue::Float64(Some(*f)))))
             }
+            Some(LiteralType::Decimal(d)) => Ok(Arc::new(Expr::Literal(ScalarValue::Decimal128(
+                Some(std::primitive::i128::from_le_bytes(
+                    d.value.clone().try_into().unwrap(),
```

Comment:
Is it OK to call `.unwrap()` here ?
Wouldn't it be better to use `?` instead and propagate the error ? The method returns `Result<Arc<Expr>>`.
IMO a library should never panic!